broker/src/asterisk/asterisk_channel_srv.erl

```diff
@@ -124,9 +138,24 @@ handle_cast({set_channels, ChannelIndex, RegistryIndex}, State = #state{config_j
   end,
   {noreply, State#state{channels = ChannelIndex, registry = RegistryIndex, config_job_state = idle}};
 
-handle_cast({set_channel_status, Status}, State = #state{channel_status = PrevStatus}) ->
+handle_cast({set_channel_status, Status}, State = #state{channel_status = PrevStatus, registration_failures = RegFailures}) ->
   channel:log_broken_channels(PrevStatus, Status),
-  {noreply, State#state{channel_status = Status, status_job_state = idle}};
+
+  % increment the registration failures
+  {NewRegFailures, DisableChannels} = update_registration_failures(Status, RegFailures),
+  if
+    length(DisableChannels) > 0 ->
+      % disable channels that have excedeed the number of failed status checks
+      lists:foreach(fun(ChannelId) ->
+                        channel:disable_by_id(ChannelId)
+                    end, DisableChannels),
```

Comment:
Two things changed my mind about this:

- I think it's a good idea to generate a log line for each disabled channel (for easier tracking)
- Automatic channel disabling shouldn't happen all that too often and it should only affect a small number of channels, due to the incremental nature of the procedure

I added a `channel:disable_by_ids/1` and moved the `lists:foreach/2` call over there though.